src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -208,6 +271,44 @@ pub fn update_lockfile(ws: &Workspace, opts: &UpdateOptions) -> CargoResult<()>
         }
         debug!("{:#?}", changes);
 
-        changes.into_iter().map(|(_, v)| v).collect()
+        let mut partly_added = BTreeMap::new();
+        let mut partly_updated = BTreeMap::new();
+        let mut partly_removed = BTreeMap::new();
+
+        {
+            let dependents_iter = changes.iter()
+                .filter(|(_, v)| v.0.len() == 0 && v.1.len() == 0)
```

Comment:
there is one edge case where this check is not enough, e.g.

1. Set up a workspace with the following members and dependencies:

```
a → bitflags="0.9"
b → bitflags="1.0"
c → bitflags="0.9"
```

2. `cargo update` to build the lock file
3. New version of bitflags is release on crates.io - `1.x`
4. Edit `c`'s version of bitflags to `1.0`
5. `cargo update` will print only the message describing an update from `1.0` to `1.x`